cmd/signature-v4-postpolicyform.go

```diff
@@ -152,19 +149,17 @@ func parsePostPolicyFormV4(policy string) (PostPolicyForm, error) {
 }
 
 // checkPostPolicy - apply policy conditions and validate input values.
-func checkPostPolicy(formValues map[string]string) APIErrorCode {
-	/// Decoding policy
-	policyBytes, err := base64.StdEncoding.DecodeString(formValues["Policy"])
-	if err != nil {
-		return ErrMalformedPOSTRequest
-	}
-	postPolicyForm, err := parsePostPolicyFormV4(string(policyBytes))
-	if err != nil {
-		return ErrMalformedPOSTRequest
-	}
+func checkPostPolicy(formValues map[string]string, postPolicyForm PostPolicyForm) APIErrorCode {
 	if !postPolicyForm.Expiration.After(time.Now().UTC()) {
 		return ErrPolicyAlreadyExpired
 	}
+	if postPolicyForm.Conditions.ContentLengthRange.Max > maxObjectSize {
+		// Upload object size can not be more than what is allowed.
+		return ErrNotImplemented
```

Comment:
aws-s3 is not responding with proper message, with (min, max) = (30, 20) and file size of 40, it gives error:
`<Message>Your proposed upload is smaller than the minimum allowed size</Message><ProposedSize>8</ProposedSize>`

what we can do is remove this check let limitReader handle the error case